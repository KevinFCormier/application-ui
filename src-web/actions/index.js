/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/*
 * action types
 *
 * action name: <NOUN>_<VERB>
 */

export const CATALOG_DROPDOWN_FILTERS_VISIBILITY_TOGGLE = 'CATALOG_DROPDOWN_FILTER_VISIBILITY_TOGGLE'
export const CATALOG_FETCH_ERROR_STATUS_CHANGE = 'CATALOG_FETCH_ERROR_STATUS_CHANGE'
export const CATALOG_INSTALL_FAILURE = 'CATALOG_INSTALL_FAILURE'
export const CATALOG_INSTALL_LOADING = 'CATALOG_INSTALL_LOADING'
export const CATALOG_INSTALL_SUCCESS = 'CATALOG_INSTALL_SUCCESS'
export const CATALOG_INSTALL_VALIDATION_FAILURE = 'CATALOG_INSTALL_VALIDATION_FAILURE'
export const CATALOG_RESOURCE_FILTER_REPOS = 'CATALOG_RESOURCE_FILTER_REPOS'
export const CATALOG_RESOURCE_FILTER_SEARCH = 'CATALOG_RESOURCE_FILTER_SEARCH'
export const CATALOG_RESOURCE_SELECT = 'CATALOG_RESOURCE_SELECT' // clicking on a tile

export const CLEAR_REQUEST_STATUS = 'CLEAR_REQUEST_STATUS'

export const CREATE_GRAFANA_DASHBOARD = 'CREATE_GRAFANA_DASHBOARD'

export const DEL_RECEIVE_FAILURE = 'DEL_RECEIVE_FAILURE'
export const DEL_RECEIVE_SUCCESS = 'DEL_RECEIVE_SUCCESS'
export const DEL_REQUEST = 'DEL_REQUEST'

export const LOGS_RESET= 'LOGS_RESET'
export const LOGS_SEARCH = 'LOGS_SEARCH'

export const MODAL_UPDATE = 'MODAL_UPDATE'

export const NAMESPACES_RECEIVE_FAILURE = 'NAMESPACES_RECEIVE_FAILURE'
export const NAMESPACES_RECEIVE_SUCCESS = 'NAMESPACES_RECEIVE_SUCCESS'

export const NAV_MODIFY = 'NAV_MODIFY'
export const NAV_RECEIVE_SUCCESS = 'NAV_RECEIVE_SUCCESS'

export const POST_RECEIVE_FAILURE = 'POST_RECEIVE_FAILURE'
export const POST_RECEIVE_SUCCESS = 'POST_RECEIVE_SUCCESS'
export const POST_REQUEST = 'POST_REQUEST'
export const PUT_RECEIVE_FAILURE = 'PUT_RECEIVE_FAILURE'
export const PUT_RECEIVE_SUCCESS = 'PUT_RECEIVE_SUCCESS'
export const PUT_REQUEST = 'PUT_REQUEST'

export const REPO_FETCH_REQUEST_SUCCESS = 'REPO_FETCH_REQUEST_SUCCESS'

export const RESOURCES_FETCH_REQUEST_LOADING = 'RESOURCE_FETCH_REQUEST_LOADING'
export const RESOURCES_FETCH_REQUEST_SUCCESS = 'RESOURCES_FETCH_REQUEST_SUCCESS'
export const RESOURCE_ADD = 'RESOURCE_ADD'
export const RESOURCE_DELETE= 'RESOURCE_DELETE'
export const RESOURCE_MODIFY = 'RESOURCE_MODIFY'
export const RESOURCE_RECEIVE_FAILURE = 'RESOURCE_RECEIVE_FAILURE'
export const RESOURCE_RECEIVE_SUCCESS = 'RESOURCE_RECEIVE_SUCCESS'
export const RESOURCE_REQUEST = 'RESOURCE_REQUEST'
export const RESOURCE_RESET= 'RESOURCE_RESET'


export const DASHBOARD_RECEIVE_SUCCESS = 'DASHBOARD_RECEIVE_SUCCESS'
export const DASHBOARD_RECEIVE_FAILURE = 'DASHBOARD_RECEIVE_FAILURE'


export const ROLE_RECEIVE_FAILURE = 'ROLE_RECEIVE_FAILURE'
export const ROLE_RECEIVE_SUCCESS = 'ROLE_RECEIVE_SUCCESS'

export const SECONDARY_HEADER_UPDATE = 'SECONDARY_HEADER_UPDATE'

export const SWITCH_NAMESPACE = 'SWITCH_NAMESPACE'

export const TABLE_PAGE_CHANGE = 'TABLE_PAGE_CHANGE'
export const TABLE_SEARCH = 'TABLE_SEARCH'
export const TABLE_SORT = 'TABLE_SORT'

export const TOPOLOGY_FILTERS_UPDATE = 'TOPOLOGY_FILTERS_UPDATE'
export const TOPOLOGY_FILTERS_REQUEST = 'TOPOLOGY_FILTERS_REQUEST'
export const TOPOLOGY_FILTERS_RECEIVE_ERROR = 'TOPOLOGY_FILTERS_RECEIVE_ERROR'
export const TOPOLOGY_FILTERS_RECEIVE_SUCCESS = 'TOPOLOGY_FILTERS_RECEIVE_SUCCESS'
export const TOPOLOGY_SELECTION_UPDATE = 'TOPOLOGY_SELECTION_UPDATE'

export const UICONFIG_RECEIVE_SUCCESS = 'UICONFIG_RECEIVE_SUCCESS'

export const UPDATE_USERPREFERENCES = 'UPDATE_USERPREFERENCES'

export const USERPREFERENCES_RECEIVE_FAILURE = 'USERPRFERENCES_RECEIVE_FAILURE'
export const USERPREFERENCES_RECEIVE_SUCCESS = 'USERPRFERENCES_RECEIVE_SUCCESS'

export const USER_LOGIN_RECEIVE_SUCCESS = 'USER_LOGIN_RECEIVE_SUCCESS'
export const USER_LOGOUT_RECEIVE_FAILURE = 'USER_LOGOUT_RECEIVE_FAILURE'
export const USER_LOGOUT_RECEIVE_SUCCESS = 'USER_LOGOUT_RECEIVE_SUCCESS'
export const USER_LOGOUT_REQUEST = 'USER_LOGOUT_REQUEST'
/*
 * other constants specific to particular actions
 */

export const VisibilityFilters = {
  SHOW_ALL: 'SHOW_ALL',
  SHOW_COMPLETED: 'SHOW_COMPLETED',
  SHOW_ACTIVE: 'SHOW_ACTIVE'
}

export const USER_LOGIN_STATUS = {
  LOGGED_IN: 'LOGGED_IN',
  LOGGED_OUT: 'LOGGED_OUT'
}

export const REQUEST_STATUS = {
  INCEPTION: 'INCEPTION',
  DONE: 'DONE',
  IN_PROGRESS: 'IN_PROGRESS',
  ERROR: 'ERROR'
}

export const PAGE_SIZES = {
  DEFAULT: 20,
  VALUES: [5, 10, 20, 50, 75, 100]
}

export const SORT_DIRECTION_ASCENDING = 'asc'
export const SORT_DIRECTION_DESCENDING = 'desc'
