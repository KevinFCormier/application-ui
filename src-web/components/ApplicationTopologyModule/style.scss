/*******************************************************************************
 * Licensed Materials - Property of IBM
 * 5737-E67
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
@import 'node_modules/carbon-components/scss/globals/scss/_colors';
@import 'scss/variables';


.resourceDiagramSourceContainer {
  position: relative;

  /* diagram and controls */
  .resourceDiagramControlsContainer {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    position: relative;
    box-shadow: 0 1px 2px 0 rgba(0,0,0,0.10);

    /* title */
    .diagram-title {
    background-color: #fff;
      font-weight: bold;
      font-size: 14px;
      border-bottom: 1px solid #DFE3E6;
      padding: 15px;
    }

    .channel-diagram-container {
      display: flex;
      height: 400px;
      box-shadow: 0 1px 2px 0 rgba(0,0,0,0.10);
      background-color: white;
      position: relative;

      .channel-controls-container {
        width: 200px;
        border-right: 1px solid #DFE3E6;
      }

      .last-update-container {
        position: absolute;
      }

      /* diagram */
      .diagramViewerDiagram {
        flex: 1;
        background-color: #fff;
        position: relative;

        .diagramViewerContainerContainer {
          width: 100%;
          height: 100%;
          background-color: #fff;
          position: relative;
          overflow: hidden;

          .secondaryLoad {
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%);
          }

          .diagramViewerContainer {
            margin-right: 0;
          }

          .bx--inline-notification--error {
            position: absolute;
            top: 16px;
            right: 16px;
          }
        }
      }
    }

    /* filter bar */
    .diagram-controls-container {
      background-color: #fff;
      display: flex;
      align-items: center;
      justify-content: space-between;
      padding: 10px 20px 10px 10px;
      border-top: 1px solid #DFE3E6;

      .diagram-type-filter-bar {
        display: flex;
      }

      .diagram-expand-button {
        display: flex;
        font-size: 12px;
        cursor: pointer;
        color: #3d70b2;

        .icon {
          margin-right: 8px;
          width: 13px;
          height: 13px;
          stroke: #3d70b2;
        }
      }
    }
  }

  /* source editor */
  .resourceEditorContainer {
    width: 100%;
    height: calc(100vh - 205px);
    display: flex;
    flex-direction: column;
    position: relative;

    .resource-editor-toolbar {
      display: flex;
      align-items: center;
    }

    .bx--inline-notification {
      position: absolute;
      top: 16px;
      left: 16px;
      z-index: 1000;
      background-color: white;
    }

    .diagram-collapse-button {
      position: absolute;
      top: 3px;
      right: 4px;
      z-index: 1000;
      font-size: 12px;
      cursor: pointer;
      color: #3d70b2;
      display: flex;
      align-items: center;

      .icon {
        margin-left: 8px;
        width: 13px;
        height: 13px;
        stroke: #3d70b2;
      }
    }
  }

  /* when editor is open */
  &.showExpandedTopology {
    height: calc(100vh - 220px);
    margin: -30px -8rem -50px -3rem;

    /* when editor is open, diagram and control positions */
    .resourceDiagramControlsContainer {
      height: calc(100vh - 225px);
      box-shadow: none;

      /* title */
      .diagram-title {
        display: none;
      }

      .channel-diagram-container {
        display: flex;
        margin-top: 26px;
        height: calc(100vh - 225px);
        box-shadow: 0 1px 2px 0 rgba(0,0,0,0.10);
        background-color: white;
        position: relative;

        .channel-controls-container {
          width: 200px;
          border-right: 1px solid #DFE3E6;
        }

        /* diagram */
        .diagramViewerDiagram {
          flex: 1;
          height: 100%;

          .diagramViewerContainerContainer {
            &.scrolled {
              overflow-y: auto;
              overflow-x: hidden;
            }
          }
        }
      }

      /* filter bar */
      .diagram-controls-container {
        position: absolute;
        display: flex;
        align-items: center;
        background-color: transparent;
        padding: 0 0 0 10px;
        border: none;

        .diagram-type-filter-bar {
          position: relative;
          bottom: unset;
          left: unset;
        }
        .diagram-expand-button {
          display: none;
        }
      }
    }
  }

  .last-update-container {
    position: absolute;
    right: 20px;
    top: 10px;
    z-index: 1000;
    display: flex;
    align-items: center;
    font-size: 12px;

    .bx--loading--small {
      width: 1rem;
      height: 1rem;
      margin: -2px 0;
    }
  }

  .channel-controls-container {
    display: flex;
    flex-direction: column;
    padding-top: 10px;

    .channel-control {
      padding: 10px 0 10px 30px;
      position: relative;
      font-size: 12px;

      &-subscripion {
      }

      &-channel {
        font-weight: bold;
      }

      &:hover {
        cursor: pointer;
        background-color: #F3F7FB;
      }

      &.selected {
        border-left: 4px solid blue;
        background-color: #F3F7FB;
      }
    }
  }


  /* when view is split, show details view on other side of diagram */
  &.split {
    .topologyDetails {
      left: -200px;
      right: unset;
    }
  }

  .yamlEditorContainer {
    height: calc(100% - 30px);
  }

  .Resizer {
    width: 0;
    opacity: .2;
    z-index: 1;
    -moz-box-sizing: border-box;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    -moz-background-clip: padding;
    -webkit-background-clip: padding;
    background-clip: padding-box;
    border-left: 3px solid rgba(255, 255, 255, 0);
    border-right: 3px solid rgba(255, 255, 255, 0);
    cursor: col-resize;
    margin: 0 5px;
  }

   .Resizer:hover {
    -webkit-transition: all .2s ease;
    transition: all .2s ease;
  }

  .Resizer.vertical:hover {
      background: rgba(0, 0, 0, 0.5);
  }

  .ace-monokai .ace_marker-layer .ace_selection {
    background-color: #808080
  }
}

