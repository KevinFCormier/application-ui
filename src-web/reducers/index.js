/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/*
 *
 * Things you should never do inside a reducer:
 *
 * - Mutate its arguments
 * - Perform side effects like API calls and routing transitions
 * - Call non-pure functions, e.g. Date.now() or Math.random()
 *
 * Reducers must be deterministic pure functions.  Given the same arguments, it should calculate the next state and return it.
 * No surprises. No side effects. No API calls. No mutations. Just a calculation.
 *
 * Selectors should sit along side reducers.
 */

import { createResourceReducer, resourceReducerFunction } from './common'
import { RESOURCE_TYPES } from '../../lib/shared/constants'
import lodash from 'lodash'

export { user, loggedIn } from './user'
export { secondaryHeader } from './common'

export { modal } from './modal'

//the exported function name must match the resourceType value
export const HCMApplicationList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_APPLICATIONS))
export const HCMClusterList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_CLUSTERS))
export const HCMComplianceList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_COMPLIANCES))
export const HCMDashboardList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_DASHBOARD))
export const HCMNodeList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_NODES))
export const HCMPersistentVolumeList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_PVS))
export const HCMPersistentVolumeClaimList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_PVS_CLAIM))
export const HCMPodList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_PODS))
export const HCMPolicyList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_POLICIES))
export const HCMReleaseList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_RELEASES))
export const HCMRepositoryList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_REPOSITORIES))

export { default as catalog } from './catalog'
export { topology } from './topology'
export { resourceFilters } from './filter'

function predicate(resourceType, action) {
  if (lodash.isEqual(resourceType, action.resourceType))
    return true
  return lodash.find(lodash.values(resourceType), type => {
    if (typeof type === 'string') {
      return type.indexOf(action.resourceType) > -1
    }
    return false
  })
}
