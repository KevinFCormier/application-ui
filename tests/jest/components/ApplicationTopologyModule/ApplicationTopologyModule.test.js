/*******************************************************************************
 * Licensed Materials - Property of Red Hat
 * Copyright (c) 2020 Red Hat, Inc.
 *******************************************************************************/

const React = require("../../../../node_modules/react");

import ApplicationTopologyModule from "../../../../src-web/components/ApplicationTopologyModule/ApplicationTopologyModule.js";

import renderer from "react-test-renderer";
import * as reducers from "../../../../src-web/reducers";
import { mount } from "enzyme";

import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import thunkMiddleware from "redux-thunk";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

const preloadedState = window.__PRELOADED_STATE__;
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const middleware = [thunkMiddleware];

const store = createStore(
  combineReducers(reducers),
  preloadedState,
  composeEnhancers(applyMiddleware(...middleware))
);

describe("ApplicationTopologyModule with selected node ID", () => {
  it("ApplicationTopologyModule renders correctly when topology is not expanded", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <ApplicationTopologyModule
              selectedNodeId={nodeID}
              showExpandedTopology={false}
              params={params}
              locale={"en-US"}
            />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("ApplicationTopologyModule renders correctly when topology is expanded", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <ApplicationTopologyModule
              selectedNodeId={nodeID}
              showExpandedTopology={true}
              params={params}
              locale={"en-US"}
            />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("ApplicationTopologyModule renders correctly when topology is expanded click", () => {
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <ApplicationTopologyModule
            selectedNodeId={nodeID}
            showExpandedTopology={true}
            params={params}
            locale={"en-US"}
          />
        </Provider>
      </BrowserRouter>
    );

    wrapper
      .find(".bx--search-input")
      .at(0)
      .simulate("change");
  });
});

const nodeID = "acb123";

const params = {
  name: "app1",
  namespace: "default"
};
