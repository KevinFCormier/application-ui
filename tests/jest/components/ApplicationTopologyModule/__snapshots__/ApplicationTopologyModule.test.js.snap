// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationTopologyModule with selected node ID ApplicationTopologyModule renders correctly when topology is expanded 1`] = `
<div
  className="resourceDiagramSourceContainer showExpandedTopology split"
>
  <div
    className="SplitPane  vertical "
    style={
      Object {
        "MozUserSelect": "text",
        "WebkitUserSelect": "text",
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "left": 0,
        "msUserSelect": "text",
        "outline": "none",
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "userSelect": "text",
      }
    }
  >
    <div
      className="Pane vertical Pane1  "
      style={
        Object {
          "flex": "none",
          "outline": "none",
          "position": "relative",
          "width": 1000,
        }
      }
    >
      <div
        className="resourceDiagramControlsContainer"
      >
        <div
          className="diagram-title"
        >
          Resource topology
        </div>
        <div
          className="diagram-controls-container"
        >
          <div
            className="diagram-type-filter-bar"
            id="type-filter-bar-portal-id"
          />
        </div>
        <div
          className="topology-container"
        >
          <div
            className="topologyDiagramContainer"
          >
            <div
              className="channel-controls-container"
              style={
                Object {
                  "height": 0,
                  "overflow": "hidden",
                  "position": "relative",
                  "width": 260,
                }
              }
            >
              <div
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "bottom": 0,
                    "left": 0,
                    "marginBottom": -17,
                    "marginRight": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <div
                  className="channel-controls-container"
                />
              </div>
              <div
                style={
                  Object {
                    "borderRadius": 3,
                    "bottom": 2,
                    "display": "none",
                    "height": 6,
                    "left": 2,
                    "position": "absolute",
                    "right": 2,
                  }
                }
              >
                <div />
              </div>
              <div
                style={
                  Object {
                    "borderRadius": 3,
                    "bottom": 2,
                    "display": "none",
                    "position": "absolute",
                    "right": 2,
                    "top": 2,
                    "width": 6,
                  }
                }
              >
                <div
                  className="channel-controls-scrollbar"
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,.2)",
                      "borderRadius": "inherit",
                      "cursor": "pointer",
                      "display": "block",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="diagramViewerDiagram"
            >
              <div
                className="diagramViewerContainerContainer"
                id="diagram-viewer-container-container"
              >
                <div
                  aria-label="zoom"
                  className="diagramViewerContainer"
                  role="region"
                  style={
                    Object {
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                >
                  <svg
                    className="topologyDiagram"
                    id="topologySvgId"
                  />
                </div>
                <div
                  className="secondaryLoad"
                >
                  <div
                    aria-live="assertive"
                    className="bx--loading"
                  >
                    <svg
                      className="bx--loading__svg"
                      viewBox="-75 -75 150 150"
                    >
                      <circle
                        cx="0"
                        cy="0"
                        r="37.5"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  className="layoutLoadingContainer"
                >
                  <div
                    aria-live="assertive"
                    className="bx--loading"
                  >
                    <svg
                      className="bx--loading__svg"
                      viewBox="-75 -75 150 150"
                    >
                      <circle
                        cx="0"
                        cy="0"
                        r="37.5"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                className="diagram-controls"
              >
                <div
                  className="zoom-buttons"
                >
                  <div
                    aria-label="Zoom to fit"
                    className="zoom-target-button"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    tabIndex="0"
                    title="Zoom to fit"
                  >
                    <svg
                      className="icon"
                    >
                      <use
                        href="#diagramIcons_zoomTarget"
                      />
                    </svg>
                  </div>
                  <div
                    aria-label="Zoom in"
                    className="zoom-in-button"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    tabIndex="0"
                    title="Zoom in"
                  >
                    <svg
                      className="icon"
                    >
                      <use
                        href="#diagramIcons_zoomIn"
                      />
                    </svg>
                  </div>
                  <div
                    aria-label="Zoom out"
                    className="zoom-out-button"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    tabIndex="0"
                    title="Zoom out"
                  >
                    <svg
                      className="icon"
                    >
                      <use
                        href="#diagramIcons_zoomOut"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      className="Resizer vertical "
      onClick={[Function]}
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    />
    <div
      className="Pane vertical Pane2  "
      style={
        Object {
          "flex": 1,
          "outline": "none",
          "position": "relative",
        }
      }
    >
      <div
        className="resourceEditorContainer"
      >
        <div
          aria-label="Type"
          className="resource-editor-toolbar"
          id="Type"
          role="region"
        >
          <div
            className="editor-bar"
          >
            <div
              aria-label="Search"
              className="editor-bar-search"
              id="Search"
              role="region"
            >
              <div
                aria-labelledby="search-name-label"
                className="bx--search bx--search--sm"
                role="search"
              >
                <svg
                  alt=""
                  aria-label=""
                  className="bx--search-magnifier"
                  fillRule="evenodd"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <title>
                    
                  </title>
                  <path
                    d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zm4.936-1.27l4.563 4.557-.707.708-4.563-4.558a6.5 6.5 0 1 1 .707-.707z"
                  />
                </svg>
                <label
                  className="bx--label"
                  htmlFor="search-name"
                  id="search-name-label"
                >
                  
                </label>
                <input
                  aria-label="Search"
                  className="bx--search-input"
                  id="search-name"
                  onChange={[Function]}
                  placeholder="Search"
                  type="text"
                />
                <button
                  className="bx--search-close bx--search-close--hidden"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    alt="Provide a description that will be used as the title"
                    aria-label="Provide a description that will be used as the title"
                    fillRule="evenodd"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <title>
                      Provide a description that will be used as the title
                    </title>
                    <path
                      d="M8 6.586L5.879 4.464 4.464 5.88 6.586 8l-2.122 2.121 1.415 1.415L8 9.414l2.121 2.122 1.415-1.415L9.414 8l2.122-2.121-1.415-1.415L8 6.586zM8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              aria-label="Previous"
              className="editor-bar-button  disabled"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
              title="Previous"
            >
              <svg
                height="16px"
                width="16px"
              >
                <use
                  href="#diagramIcons_previous"
                />
              </svg>
            </div>
            <div
              aria-label="Next"
              className="editor-bar-button  disabled"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
              title="Next"
            >
              <svg
                height="16px"
                width="16px"
              >
                <use
                  href="#diagramIcons_next"
                />
              </svg>
            </div>
            <div
              className="editor-bar-spacer"
            />
            <div
              aria-label="Undo"
              className="editor-bar-button  disabled"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
              title="Undo"
            >
              <svg
                height="16px"
                width="16px"
              >
                <use
                  href="#diagramIcons_undo"
                />
              </svg>
            </div>
            <div
              aria-label="Redo"
              className="editor-bar-button  disabled"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
              title="Redo"
            >
              <svg
                height="16px"
                width="16px"
              >
                <use
                  href="#diagramIcons_redo"
                />
              </svg>
            </div>
            <div
              aria-label="Restore"
              className="editor-bar-button  disabled"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
              title="Restore"
            >
              <svg
                height="16px"
                width="16px"
              >
                <use
                  href="#diagramIcons_restore"
                />
              </svg>
            </div>
            <div
              className="editor-bar-spacer"
            />
            <div
              aria-label="Update"
              className="editor-bar-button  disabled"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
              title="Update"
            >
              <svg
                height="16px"
                width="16px"
              >
                <use
                  href="#diagramIcons_deploy"
                />
              </svg>
            </div>
          </div>
        </div>
        
        <div
          className="yamlEditorContainer"
        >
          <div
            id="brace-editor"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </div>
        <div
          aria-label="Close editor"
          className="diagram-collapse-button"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
          title="Close editor"
        >
          Close editor
          <svg
            className="icon"
          >
            <use
              href="#diagramIcons_launch"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ApplicationTopologyModule with selected node ID ApplicationTopologyModule renders correctly when topology is not expanded 1`] = `
<div
  className="resourceDiagramSourceContainer"
>
  <div
    className="resourceDiagramControlsContainer"
  >
    <div
      className="diagram-title"
    >
      Resource topology
    </div>
    <div
      className="topologyDiagramContainer"
    >
      <div
        className="channel-controls-container"
        style={
          Object {
            "height": 0,
            "overflow": "hidden",
            "position": "relative",
            "width": 260,
          }
        }
      >
        <div
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "bottom": 0,
              "left": 0,
              "marginBottom": -17,
              "marginRight": 0,
              "overflow": "scroll",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <div
            className="channel-controls-container"
          />
        </div>
        <div
          style={
            Object {
              "borderRadius": 3,
              "bottom": 2,
              "display": "none",
              "height": 6,
              "left": 2,
              "position": "absolute",
              "right": 2,
            }
          }
        >
          <div />
        </div>
        <div
          style={
            Object {
              "borderRadius": 3,
              "bottom": 2,
              "display": "none",
              "position": "absolute",
              "right": 2,
              "top": 2,
              "width": 6,
            }
          }
        >
          <div
            className="channel-controls-scrollbar"
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,.2)",
                "borderRadius": "inherit",
                "cursor": "pointer",
                "display": "block",
                "position": "relative",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <div
        className="diagramViewerDiagram"
      >
        <div
          className="diagramViewerContainerContainer"
          id="diagram-viewer-container-container"
        >
          <div
            aria-label="zoom"
            className="diagramViewerContainer"
            role="region"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <svg
              className="topologyDiagram"
              id="topologySvgId"
            />
          </div>
          <div
            className="secondaryLoad"
          >
            <div
              aria-live="assertive"
              className="bx--loading"
            >
              <svg
                className="bx--loading__svg"
                viewBox="-75 -75 150 150"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="37.5"
                />
              </svg>
            </div>
          </div>
          <div
            className="layoutLoadingContainer"
          >
            <div
              aria-live="assertive"
              className="bx--loading"
            >
              <svg
                className="bx--loading__svg"
                viewBox="-75 -75 150 150"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="37.5"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="diagram-controls"
        >
          <div
            className="zoom-buttons"
          >
            <div
              aria-label="Zoom to fit"
              className="zoom-target-button"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
              title="Zoom to fit"
            >
              <svg
                className="icon"
              >
                <use
                  href="#diagramIcons_zoomTarget"
                />
              </svg>
            </div>
            <div
              aria-label="Zoom in"
              className="zoom-in-button"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
              title="Zoom in"
            >
              <svg
                className="icon"
              >
                <use
                  href="#diagramIcons_zoomIn"
                />
              </svg>
            </div>
            <div
              aria-label="Zoom out"
              className="zoom-out-button"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
              title="Zoom out"
            >
              <svg
                className="icon"
              >
                <use
                  href="#diagramIcons_zoomOut"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="diagram-controls-container"
    >
      <div
        className="diagram-type-filter-bar"
        id="type-filter-bar-portal-id"
      />
      <div
        aria-label="Edit topology"
        className="diagram-expand-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
        title="Edit topology"
      >
        <svg
          className="icon"
        >
          <use
            href="#diagramIcons_launch"
          />
        </svg>
        Edit topology
      </div>
    </div>
  </div>
</div>
`;
